test/models/submission_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class SubmissionTest < ActiveSupport::TestCase
+  test "submission requires a user_id" do
+    sub_w_user = build(:text_submission)
+    sub_no_user = build(:text_submission, user: nil)
+    assert sub_w_user.valid?
+    assert_not sub_no_user.valid?
+  end
+
+  test "submission with url and title is valid" do
+    sub = build(:url_submission)
+    assert sub.valid?
+  end
+
+  test "submission with text is valid" do
+    sub = build(:text_submission)
+    assert sub.valid?
+  end
```

Comment:
these 2 tests are just testing that the factory definitions are setup properly. while there's some merit to that, I think I'd rather see specific attributes set in the `build` method and assert that those records are valid.

Reply:
True.... it made more sense to have these when I first started writing the tests.